Fix links in "See also" section of the API docs.

ParmarKrishna opened the pull request:
@garg3133 Review Requested.

I have been investigating to find any hard-coded redirect but there are none and entire docs have been updated with `/guide`.

Afterwards, I found that there is if-else logic running for "See Also" section at, `nightwatch-docs/src/includes/api-method.ejs`

In that, the line, `<li><code><a href="<%- appSettings.baseUrl.replace(/\/$/, '') -%>/api/<%- link.split('.').join('/') %>.html"><%= link %></a></code></li>` generates link accordingly.

Now, after many workarounds, I cannot seem to budge the extra two dots that, `appSettings.baseUrl.replace(/\$/,'')` is appending after ".org" domain.

I tried:
[-] Slicing after replace with (0,-2), but instead of removing double dots, it gives out: https://nightwatchjs.o../<other-parts-of-url>
[-] Writing new logic: Gives out same result

I found that, replace() method is creating the problems. Simply removing it, works. Additionally, I added "/" in the BASE_URL in postdoc config.

Do tell me if this is not right, or should I seek another way to solve the problem?

ParmarKrishna committed:
#289 Bug Fixed

@garg3133 Review Requested.

I have been investigating to find any hard-coded redirect but there are none and entire docs have been updated with `/guide`.

Afterwards, I found that there is if-else logic running for "See Also" section at, `nightwatch-docs/src/includes/api-method.ejs`

In that, the line, `<li><code><a href="<%- appSettings.baseUrl.replace(/\/$/, '') -%>/api/<%- link.split('.').join('/') %>.html"><%= link %></a></code></li>` generates link accordingly.

Now, after many workarounds, I cannot seem to budge the extra two dots that, `appSettings.baseUrl.replace(/\$/,'')` is appending after ".org" domain.

I tried:
[-] Slicing after replace with (0,-2), but instead of removing double dots, it gives out: https://nightwatchjs.o../<other-parts-of-url>
[-] Writing new logic: Gives out same result

I found that, replace() method is creating the problems. Simply removing it, works. Additionally, I added "/" in the BASE_URL in postdoc config.

Do tell me if this is not right, or should I seek another way to solve the problem?

garg3133 commented:
@ParmarKrishna Your solution does not seem to be working for me. Maybe you can try removing the complete baseUrl itself from the `href` attribute and just start the link with `/api/`?

ParmarKrishna committed:
BugFix: remove baseUrl

@garg3133 Review Requested.

As per your suggestion, removing entire appSettings.baseUrl works. Pages seem to be redirected with directly `api`

ParmarKrishna commented:
@garg3133 Review Requested.

As per your suggestion, removing entire baseUrl does work fine, for all the "See Also" sections.

I did:
- Remove `baseUrl` from the line `<li><code><a href="<%- appSettings.baseUrl.replace(/\/$/, '') -%>/api/<%- link.split('.').join('/') %>.html"><%= link %></a></code></li>` at, `nightwatch-docs/src/includes/api-method.ejs`

It works. Do let me know if this solution works for you, or should I look for other ways?

PR #291 has been pushed with the changes.

garg3133 commented on postdoc.config.js:
Can you revert all the changes in this file?

ParmarKrishna replied:
Understood. On it.

ParmarKrishna replied:
I have revert all the changes.

garg3133 replied:
There's still a change in that file. We ideally shouldn't have any unnecessary files in a PR.

ParmarKrishna replied:
Hey @garg3133,

I do not see any other changes besides remove of "/" in BaseURL. Please clarify which other changes are you referring to.

garg3133 replied:
What I meant is that the `postdoc.config.js` file should not be a part of this PR (it should not be present in the "Files changed" tab of this PR).

I've removed the file myself now.

ParmarKrishna replied:
oh okay. I apologise as I am new to the opensource, I wasn't aware.

garg3133 reviewed (commented):
Looks good. Can you remove all the unnecessary changes from the PR?

kp250322 committed:
Revert Changes at postdoc config

kp250322 committed:
Revert "Revert Changes at postdoc config"

This reverts commit 36ceef016e2a2023e2df740f3167eb937454ee2e.

ParmarKrishna committed:
Revert changes at postdoc config

ParmarKrishna committed:
Revert changes at postdoc config

Adding the empty line at, line 19.

garg3133 committed:
Revert postdoc.config.js changes.

garg3133 commented:
The PR title should contain a brief on what this PR does instead of just writing the issue number.

garg3133 reviewed (approved):
Looks good!
